Author: Daniel Baumann <daniel.baumann@progress-technologies.net>
Description: Correcting charset argument when calling locale-gen in debian template (Closes: #607273).

diff -Naurp lxc.orig/templates/lxc-debian.in lxc/templates/lxc-debian.in
--- lxc.orig/templates/lxc-debian.in	2010-12-19 17:12:57.299842657 +0100
+++ lxc/templates/lxc-debian.in	2010-12-19 17:14:44.302778785 +0100
@@ -74,10 +74,10 @@ EOF
 
     # reconfigure some services
     if [ -z "$LANG" ]; then
-	chroot $rootfs locale-gen en_US.UTF-8
+	chroot $rootfs locale-gen en_US.UTF-8 UTF-8
 	chroot $rootfs update-locale LANG=en_US.UTF-8
     else
-	chroot $rootfs locale-gen $LANG
+	chroot $rootfs locale-gen $LANG $(echo $LANG | cut -d. -f2)
 	chroot $rootfs update-locale LANG=$LANG
     fi
 
